persistent/Database/Persist/TH.hs

```diff
@@ -929,6 +1153,12 @@ data MkPersistSettings = MkPersistSettings
     -- ^ TODO: document
     --
     -- @since 2.13.0.0
+    , mpsCreateDataType :: !Bool
+    -- ^ Create data type for the entity
+    , mpsFieldSqlType :: FieldDef -> SqlTypeExp
+    , mpsRecordFieldToHaskellName :: Text -> Text -> Text
+    -- ^ Derive the Haskell names using the entity and field name. The result
+    -- should be a valid haskell indentifier (start with an lower cased letter).
```

Comment:
I don't think I understand the meaning of this option from the comments.

The default doesn't alter it at all, and the `derivePersist` stuff uses `stripEntityNamePrefix`. The intent is to keep the `FieldNameHS` behavior consistent, so that you don't get the entity name prefix on a derived record.

My guess is that the intent here is to allow folks to use eg `DuplicateRecordFields` or similar and *not* have the type name prefix.

Reply:
OK, since this flag is only used by the `derivePersist` functions, I don't think we need it - it can be accepted by `derivePersist` or in a `DerivePersistSettings` record.